pkg/haprovider/haprovider.go

```diff
@@ -123,6 +122,45 @@ func (r *HAProvider) createService(
 	return service, err
 }
 
+func (r *HAProvider) getServiceAnnotation(ctx context.Context, cluster *clusterv1.Cluster) map[string]string {
```

Comment:
consider refactoring this function? it looks like it does more than reading and retrieving `annotation`? maybe move the following handling and checking outside?

Reply:
I prefer to keep it like this, because everything is in this function is for generating annotation. Listing adc and getting AviInfraSetting are what we need to generating annotation.